examples/aptos-friend/frontend/components/TradeShare.tsx

```diff
@@ -0,0 +1,104 @@
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import { useState } from "react";
+import { useWalletClient } from "@thalalabs/surf/hooks";
+
+import { Button } from "@/components/ui/button";
+import { ABI } from "@/utils/abi";
+import { Card, CardHeader, CardTitle, CardContent } from "@/components/ui/card";
+import { Toggle } from "@/components/ui/toggle";
+import { Label } from "@/components/ui/label";
+import { Input } from "@/components/ui/input";
+import { useGetHolding, useGetIssuerObjectAddress } from "@/hooks/useHolding";
+import { aptosClient } from "@/utils/aptosClient";
+import { useToast } from "@/components/ui/use-toast";
+
+type TradeShareProps = {
+  issuerAddress: `0x${string}`;
+};
+
+export function TradeShare({ issuerAddress }: TradeShareProps) {
+  const { client: walletClient } = useWalletClient();
+  const { account } = useWallet();
+  const { toast } = useToast();
+  const holding = useGetHolding(issuerAddress, account?.address as `0x${string}`);
+  const issuerObjectAddress = useGetIssuerObjectAddress(issuerAddress);
+
+  const [amount, setAmount] = useState(0);
+  const [isBuying, setIsBuying] = useState(true);
+
+  const handleTradeSubmit = async () => {
+    if (!walletClient || !issuerObjectAddress) return;
+
+    let resp;
+    if (isBuying) {
+      resp = await walletClient.useABI(ABI).buy_share({
+        type_arguments: [],
```

Comment:
i think you should be able to not include it if it is empty
```suggestion
       
```

Reply:
this is surf's requirement